integration-tests/src/test/java/tachyon/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+  public static void beforeClass() {
+    HeartbeatContext.setTimerClass(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING,
+        HeartbeatContext.SCHEDULED_TIMER_CLASS);
+    HeartbeatContext.setTimerClass(HeartbeatContext.WORKER_LINEAGE_SYNC,
+        HeartbeatContext.SCHEDULED_TIMER_CLASS);
+  }
+
+  @Before
+  public void before() throws Exception {
+    mJob = new CommandLineJob("test", new JobConf("output"));
+    mTestConf = mLocalTachyonClusterResource.get().getMasterTachyonConf();
+  }
+
+  @Test
+  public void lineageCreationTest() throws Exception {
+    LineageMasterClient lineageMasterClient = getLineageMasterClient();
+
+    try {
+      lineageMasterClient.createLineage(Lists.<String>newArrayList(), Lists.newArrayList(OUT_FILE),
+          mJob);
+
+      List<LineageInfo> infos = lineageMasterClient.getLineageInfoList();
+      Assert.assertEquals(1, infos.size());
+      Assert.assertEquals(LineageFileState.CREATED.toString(),
+          infos.get(0).outputFiles.get(0).state);
+    } finally {
+      lineageMasterClient.close();
+    }
+  }
+
+  @Test
+  public void lineageCompleteAndAsyncPersistTest() throws Exception {
+    LineageMasterClient lineageMasterClient = getLineageMasterClient();
+
+    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING, 5,
+        TimeUnit.SECONDS));
+    Assert.assertTrue(
+        HeartbeatScheduler.await(HeartbeatContext.WORKER_LINEAGE_SYNC, 5, TimeUnit.SECONDS));
+
+    try {
+      lineageMasterClient.createLineage(Lists.<String>newArrayList(), Lists.newArrayList(OUT_FILE),
+          mJob);
+
+      OutStreamOptions options = new OutStreamOptions.Builder(mTestConf)
+          .setWriteType(WriteType.MUST_CACHE).setBlockSizeBytes(BLOCK_SIZE_BYTES).build();
+      TachyonLineageFileSystem tfs =
+          (TachyonLineageFileSystem) mLocalTachyonClusterResource.get().getClient();
+      FileOutStream outputStream = tfs.getOutStream(new TachyonURI(OUT_FILE), options);
+      outputStream.write(1);
+      outputStream.close();
+
+      List<LineageInfo> infos = lineageMasterClient.getLineageInfoList();
+      Assert.assertEquals(LineageFileState.COMPLETED.toString(),
+          infos.get(0).outputFiles.get(0).state);
+
+      // Execute the checkpoint scheduler for async checkpoint
+      HeartbeatScheduler.schedule(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING);
+      HeartbeatScheduler.schedule(HeartbeatContext.WORKER_LINEAGE_SYNC);
+      Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_CHECKPOINT_SCHEDULING, 5,
+          TimeUnit.SECONDS));
+      Assert.assertTrue(
+          HeartbeatScheduler.await(HeartbeatContext.WORKER_LINEAGE_SYNC, 5, TimeUnit.SECONDS));
+
+      infos = lineageMasterClient.getLineageInfoList();
+      Assert.assertEquals(LineageFileState.PERSISENCE_REQUESTED.toString(),
+          infos.get(0).outputFiles.get(0).state);
+
+      // sleep and wait for worker to persist the file
+      CommonUtils.sleepMs(5);
```

Comment:
Correct. That thread is not managed by heartbeat context.
I think 5ms should be long enough.